Add basic Lerna monorepo support

m-mitchell opened the pull request:
Adds support for building lerna monorepo projects. 

To test:
- In your video platform working directory, make sure `origin/master` points to a version of the video platform that has the release config added at https://github.com/HippoEducation/video-platform-client/pull/310 . An explanation on how to do this without merging that PR is over there, or you can merge it. :) 
- In your package-release working directory, check out this PR and install (`composer install`). 
- From `/so/video-platform-client/work-**NAME**` run:
```
export PUBLIC_URL=/video-platform-client/work-**NAME**/packages/web/build
export REACT_APP_HOST=https://hippostage.silverorange.com
export REACT_APP_SIGN_IN_URL=https://hippostage.silverorange.com/hippo-ed/work-hive/dist/signin
export REACT_APP_CART_CALLBACK_URI=https://hippostage.silverorange.com/video-platform-client/stage/packages/web/build
export REACT_APP_GA_ACCOUNT=UA-152638193-01
export REACT_APP_GRAPHQL_URI=https://hippostage-api.silverorange.com/graphql
export REACT_APP_JSONAPI_URI=https://hippostage-api.silverorange.com
export REACT_APP_STAGE_SERVER=https://hippostage.silverorange.com
export REACT_APP_STAGE_DIR=work-meg
export REACT_APP_MEDIA_CDN="https://hippo-med-stage.s3.amazonaws.com"
export REACT_APP_CART_URI=http://localhost:6006
export REACT_APP_CART_CALLBACK_URI=https://hippostage.silverorange.com
export REACT_APP_GA_ACCOUNT=UA-152638193-01
export REACT_APP_FRIENDBUY_ACCOUNT=site-1f530c77-www.hippoeducation.com
export REACT_APP_FIREBASE_API_KEY=AIzaSyDaeVS8DrCaL7Evo0qC8_sw5I7tcV1yeXk
export REACT_APP_FIREBASE_AUTH_DOMAIN=hippo-development.firebaseapp.com
export REACT_APP_FIREBASE_PROJECT_ID=hippo-development
export REACT_APP_FIREBASE_STORAGE_BUCKET=hippo-development.appspot.com
export REACT_APP_FIREBASE_MESSAGING_SENDER_ID=306782222008
export REACT_APP_FIREBASE_APP_ID=1:306782222008:web:d8591660d13aba6962264d
/so/packages/package_release/work-**NAME**/bin/prepare-site
```

This should create a **working** web build in your working directory accessible at https://hippostage.silverorange.com/video-platform-client/work-**NAME**/packages/web/build/  

The env vars are set up in order to test that the built code is working without syncing it to prod. When preparing the site for prod, we should just run `prepare-site`; ansible will set up the correct environment variables for us.

Commit:
Add basic Lerna monorepo support

m-mitchell commented on src/Console/Command/PrepareSiteCommand.php:
```diff
@@ -143,14 +144,14 @@ protected function execute(
-            ]);
-            return 1;
-        }
+        // if (!$this->isMonoRepo() && !$this->isInLiveDirectory()) {
```
This bit needs to be uncommented before merge. Currently, it's just here so that we can test prepare-site without being in the live directory.

jaymefSO commented on src/Builder/LernaBuilder.php:
```diff
@@ -0,0 +1,40 @@
+
+    public function build(OutputInterface $output): bool
+    {
+        return Npm::install($output)
```
I believe the npm install is redundant and dep install would be covered by bootstrap.

Also, do we are skipping building native but is. there a way we could also skip installing its dependencies as well?

Commit:
Improve Lerna release process:

1. Split builds into separate commands and run them
in the order they appear in the metadata.
2. Remove redundant yarn install (this is handled by Lerna)
3. Skip installation of native package dependencies

m-mitchell commented:
I have opened a new PR here to tweak the ordering of the release metadata so we can control build order: https://github.com/HippoEducation/video-platform-client/pull/310

jaymefSO commented on src/Tool/Lerna.php:
```diff
@@ -0,0 +1,57 @@
+
+    public static function bootstrap(OutputInterface $output, Array $scopes): bool
+    {
+        $command = 'yarn lerna bootstrap';
```
this should just be `lerna bootstrap` not `yarn lerna bootstrap`

m-mitchell replied:
`lerna bootstrap` didn't work during testing since it wasn't globally installed, but since it is now I will update the PR 👍

jaymefSO commented on src/Tool/Lerna.php:
```diff
@@ -0,0 +1,57 @@
+{
+    public static function verify(OutputInterface $output): bool
+    {
+        $command = 'yarn global add lerna';
```
we shouldn't need to do this anymore. The lerna package is now installed system-wide on dev servers

Commit:
Rely on globally-installed Lerna version

Commit:
Remove test comments
